### PR TITLE
Minor doc update

### PR DESCRIPTION
**Pull request by acoates-ms**
Found an old enlistment where I updated some docs that never made it into the react-native-windows-samples repo.  

Not sure the best way to verify what these docs markdown looks like.  -- But figure its better with something than nothing.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7085)

### CODE REVIEW

**Review by NickGerleman — APPROVED** [review]
Would it be useful to add a quick blurb on thread affinity of event? I remember being thrown off by that first time using API.

File: vnext/Microsoft.ReactNative/ReactInstanceSettings.idl [hunk 1]
```diff
@@ -140,7 +140,17 @@ Subsequent runs of the application should be faster as the JavaScript will be lo
     UInt16 SourceBundlePort { get; set; };
 
     JSIEngine JSIEngineOverride { get; set; };
+
+    DOC_STRING("This event is triggered when a React Native instance has been created, but before any JavaScript is loaded in the JavaScript engine.  The `Context` Property on the event arguments provides access to the instance.")
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/ReactInstanceSettings.idl:144]:
```suggestion
    DOC_STRING("This event is triggered when a React Native instance has been created, but before any JavaScript is loaded in the JavaScript engine.  The @InstanceCreatedEventArgs.Context property on the event arguments provides access to the instance.")
```

File: vnext/Microsoft.ReactNative/ReactInstanceSettings.idl [hunk 2]
```diff
@@ -140,7 +140,17 @@ Subsequent runs of the application should be faster as the JavaScript will be lo
...
     event Windows.Foundation.EventHandler<InstanceCreatedEventArgs> InstanceCreated;
+
+    DOC_STRING("This event is triggered when React Native instance has finished loading the JavaScript bundle.  If there were errors, then the `Failed` property on the args will be true.\n\
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/ReactInstanceSettings.idl:147]:
```suggestion
    DOC_STRING("This event is triggered when React Native instance has finished loading the JavaScript bundle.  If there were errors, then the @InstanceLoadedEventArgs.Failed property on the args will be true.\n\
```

File: vnext/Microsoft.ReactNative/ReactInstanceSettings.idl [hunk 3]
```diff
@@ -140,7 +140,17 @@ Subsequent runs of the application should be faster as the JavaScript will be lo
...
+\n\
+Error types include:\n\
+* JavaScript Syntax Errors\n\
+* Global JavaScript Exception Thrown\n\
+\n\
+The `Context` Property on the event arguments provides access to the instance.")
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/ReactInstanceSettings.idl:153]:
```suggestion
The @InstanceLoadedEventArgs.Context Property on the event arguments provides access to the instance.")
```

**Review by asklar — APPROVED** [review]

**Review by jonthysell — APPROVED** [review]

**Comment by asklar** [comment]
@acoates-ms should we backport this to 64?

**Pull request merged**